src/engine/evaluate-patient.ts

```diff
@@ -67,21 +79,30 @@ function processCQLCommon(patient: Bundle, cql: string): Promise<PatientData> {
         return convertBasicCQL(cql);
       }
     })
-    .then(elm => {
-      let elmResults: ElmResults = {
-        patientResults: {}
-      };
-      if (instanceOfElmObject(elm)) {
-        elmResults = executeElm(patient, elm.main, elm.libraries);
-      } else {
-        elmResults = executeElm(patient, elm);
-      }
+    .then(elm => processELMCommon(patient, elm));
+}
 
-      // TODO - update pathwaysData to take multiple patients
-      const patientIds = Object.keys(elmResults.patientResults);
-      const patientData = elmResults.patientResults[patientIds[0]];
-      return patientData;
-    });
+/**
+ * Common logic to execute the given ELM against the given Patient.
+ * @param patient - Patient's record as FHIR data
+ * @param elm - resulting ELM from converting the CQL in a pathway
+ * @return the raw, unprocessed patientResults
+ *         derived from executing the ELM against the given patient
+ */
+function processELMCommon(patient: Bundle, elm: object): PatientData {
```

Comment:
Also this probably should have been addressed in my R4 typing PR, but having `patient` be the variable name for what is actually a bundle of the patient's record is pretty confusing.  It should be updated at some point.

Reply:
What do you want to call it? From the type information we know it's a Bundle, and it's not supposed to be any arbitrary bundle, but a single patient's entire record.

Reply:
I'd suggest renaming to patientRecord.

Reply:
Renamed to patientRecord